ompi/mpi/java/c/Makefile.am

```diff
@@ -14,7 +16,7 @@
 if OMPI_WANT_JAVA_BINDINGS
 
 # Get the include files that were generated from the .java source files
-AM_CPPFLAGS = -I$(top_builddir)/ompi/mpi/java/java $(OPAL_JDK_CPPFLAGS) -DOPAL_DYN_LIB_SUFFIX=\"$(OPAL_DYN_LIB_SUFFIX)\"
+AM_CPPFLAGS = -I$(top_builddir)/ompi/mpi/java/java $(OPAL_JDK_CPPFLAGS) -DOPAL_DYN_LIB_SUFFIX=\"$(OPAL_DYN_LIB_SUFFIX)\" -DOMPI_LIBDIR=\"$(OMPI_WRAPPER_LIBDIR)\"
```

Comment:
You know, I can't think of a reason why libmpijava is not linked against libmpi (!).  That *may* actually solve the problem described in the comment (that java will dlopen libmpi in a private namespace) -- but I think you'll need to make at least one code path from libmpijava to call into libmpi to make that linkage work.

If that works, when `libmpijava.so` is loaded, it'll pull `libmpi.so` with it, and we won't need the `dlopen()`.

As for the other ideas (if linking against libmpi doesn't work out): 

* trying to dlopen `$OPAL_PREFIX...` is pretty much what installdirs does (you'd also need to check `$OPAL_LIBDIR`, IIRC).  I'm a bit hesitant to basically repeat (some of) that logic up here.  Hmm.
* I think you're right that we should be able to assume that libmpijava is in the same dir as libmpi.  So if you can find some way to discover that directory (even if it's an OS X-specific way), that might be a good way to go.